web/swagger/api.yaml

```diff
@@ -0,0 +1,817 @@
+            the request
+      security:
+      - api-key: []
+      summary: Server settings
+      tags:
+      - Settings
+  /v0/trading-fees:
+    get:
+      description: Get trading fees of DEX.
+      produces:
+      - application/json;charset=utf-8
+      responses:
+        '200':
+          description: ''
+          schema:
+            $ref: '#/definitions/TradingFees'
+        '401':
+          description: Unauthorized access - API key missing
+        '403':
+          description: Forbidden - The API key does not have permission to perform
+            the request
+      security:
+      - api-key: []
+      summary: Trading fees
+      tags:
+      - Trading Fees
+  /v0/tx/sign:
+    post:
+      consumes:
+      - application/json;charset=utf-8
+      description: Signs the given transaction using key configured in server.
+      parameters:
+      - in: body
+        name: body
+        required: true
+        schema:
+          $ref: '#/definitions/GYTx'
+      produces:
+      - application/json;charset=utf-8
+      responses:
+        '200':
+          description: ''
+          schema:
+            $ref: '#/definitions/GYTx'
+        '400':
+          description: Invalid `body`
+        '401':
+          description: Unauthorized access - API key missing
+        '403':
+          description: Forbidden - The API key does not have permission to perform
+            the request
+      security:
+      - api-key: []
+      summary: Sign a transaction
+      tags:
+      - Transaction
+  /v0/tx/sign-and-submit:
+    post:
+      consumes:
+      - application/json;charset=utf-8
+      description: Signs the given transaction using key configured in server and
+        submits it to the network.
+      parameters:
+      - in: body
+        name: body
+        required: true
+        schema:
+          $ref: '#/definitions/GYTx'
+      produces:
+      - application/json;charset=utf-8
+      responses:
+        '200':
+          description: ''
+          schema:
+            $ref: '#/definitions/GYTxId'
+        '400':
+          description: Invalid `body`
+        '401':
+          description: Unauthorized access - API key missing
+        '403':
+          description: Forbidden - The API key does not have permission to perform
+            the request
+      security:
+      - api-key: []
+      summary: Sign and submit a transaction
+      tags:
+      - Transaction
+  /v0/tx/submit:
+    post:
+      consumes:
+      - application/json;charset=utf-8
+      description: Submits the given transaction to the network.
+      parameters:
+      - in: body
+        name: body
+        required: true
+        schema:
+          $ref: '#/definitions/GYTx'
+      produces:
+      - application/json;charset=utf-8
+      responses:
+        '200':
+          description: ''
+          schema:
+            $ref: '#/definitions/GYTxId'
+        '400':
+          description: Invalid `body`
```

Comment:
Please consider creating a generic schema for error responses like this one:

https://github.com/geniusyield/market-maker/blob/86be53efe6cea02fd174456f5821d172c7ae6b74/openapi.yaml#L31-L54

Please also add the HTTP 500 status code as well as potential response

Reply:
I have added the 500 response code. Generic schema would require manual plumbing and I would suggest to have it at a later point if it's not a blocker.